primefaces/src/main/java/org/primefaces/component/toolbar/ToolbarRenderer.java

```diff
@@ -115,4 +119,19 @@ public void encodeChildren(FacesContext facesContext, UIComponent component) thr
     public boolean getRendersChildren() {
         return true;
     }
+
+    protected boolean shouldBeRendered(FacesContext facesContext, Toolbar toolbar) {
+        if (toolbar.getChildCount() > 0) {
+            for (UIComponent child : toolbar.getChildren()) {
+                if (child.isRendered() && child instanceof ToolbarGroup) {
+                    ToolbarGroup toolbarGroup = (ToolbarGroup) child;
+                    return toolbarGroup.getChildren().stream().filter(me -> !(me instanceof Separator)).anyMatch(UIComponent::isRendered);
```

Comment:
shouldnt it use
`shouldBeRendered(FacesContext context, MenuElement menuElement)`
because push/pop?

Reply:
Theoretically yes, but since Toolbar is the only component that is no real Menu, it does not extend BaseMenuRenderer and does not use MenuItems to define it's children, I added a similiar method that's based around UIComponent to ToolbarRenderer itself.